SQL assignment.txt

```diff
@@ -0,0 +1,1533 @@
+
+
+
+8) Count the orders of each of the salespeople and output the results in descending order.
+
+mysql> SELECT salespeople.snum, salespeople.sname, COALESCE(COUNT(orders.onum), 0) as order_count  FROM salespeople  LEFT JOIN orders ON salespeople.snum = orders.snum GROUP BY salespeople.snum, salespeople.sname ORDER BY order_count DESC;
+
++------+---------+-------------+
+| snum | sname   | order_count |
++------+---------+-------------+
+| 1001 | Peel    |           3 |
+| 1002 | serres  |           3 |
+| 1007 | rifkin  |           2 |
+| 1003 | axeirod |           1 |
+| 1004 | motika  |           1 |
+| 1008 | fran    |           0 |
++------+---------+-------------+
+
+
+
+9) List the customer table if and only if one or more of the customers in the Customer table are located in SanJose.
+
+mysql> select * from customers WHERE EXISTS (SELECT city FROM customers where city="sanjose");
+
++------+----------+---------+--------+------+
+| cnum | cname    | city    | rating | snum |
++------+----------+---------+--------+------+
+| 2001 | hoffman  | london  |    100 | 1001 |
+| 2002 | giovamni | rome    |    200 | 1003 |
+| 2003 | liu      | sanjose |    200 | 1002 |
+| 2004 | grass    | berlin  |    300 | 1002 |
+| 2006 | cisneras | london  |    100 | 1001 |
+| 2006 | cisneras | sanjose |    300 | 1007 |
+| 2007 | pereira  | rome    |    100 | 1004 |
++------+----------+---------+--------+------+
+
+
+
+10) Match salespeople to customers according to what city they live in.
+
+mysql>  SELECT salespeople.* FROM salespeople LEFT JOIN customers ON customers.city = salespeople.city;
+
++------+---------+-----------+------+
+| snum | sname   | city      | comm |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1001 | Peel    | London    |   12 |
+| 1002 | serres  | sanjose   |   13 |
+| 1002 | serres  | sanjose   |   13 |
+| 1003 | axeirod | new york  |   10 |
+| 1004 | motika  | london    |   11 |
+| 1004 | motika  | london    |   11 |
+| 1007 | rifkin  | barcelona |   15 |
+| 1008 | fran    | london    |   25 |
+| 1008 | fran    | london    |   25 |
++------+---------+-----------+------+
+
+
+
+11) Find all the customers in SanJose who have a rating above 200. 
+
+mysql> select * from customers WHERE city="sanjose" and rating > 200;
+
++------+----------+---------+--------+------+
+| cnum | cname    | city    | rating | snum |
++------+----------+---------+--------+------+
+| 2006 | cisneras | sanjose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+
+
+12) List the names and commissions of all salespeople in London.
+
+mysql> select * from salespeople where city="london";
+
++------+--------+--------+------+
+| snum | sname  | city   | comm |
++------+--------+--------+------+
+| 1001 | Peel   | London |   12 |
+| 1004 | motika | london |   11 |
+| 1008 | fran   | london |   25 |
++------+--------+--------+------+
+
+
+
+13) List all the orders of Salesperson Motika from the orders table.
```

Comment:
instead of writing snum of motika by yourself , Write subquery for that.

Reply:
resolved